spec/lib/state_machine_spec.rb

```diff
@@ -126,10 +126,25 @@
     expect(test.make_second_state).to be true
     expect(test.second_state_at).to eq(was)
   end
+
+  it "test overwriting ids" do
+    test = StateMachineTestClass.create
+    expect(test.second_state_by_id).to be_nil
+
+    expect(test.make_second_state(2)).to be true
+
+    was = test.second_state_by_id
+    test.overwrite_second_state_by_id = false
```

Comment:
perhaps set it up like we do in wantable-2
```ruby
  cattr_accessor :overwrite_billing_failed_at
  self.overwrite_billing_failed_at = false
```
?

Reply:
I set this test up like we did for the overwrite_second_state_at, which is working. Are you cool with that? or do you want me to change it all up?

Reply:
nah. w/e it cool